flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/BroadcastingOutputCollector.java

```diff
@@ -31,12 +32,19 @@
 
 class BroadcastingOutputCollector<T> implements WatermarkGaugeExposingOutput<StreamRecord<T>> {
 
+    protected final OutputWithRecordsCountCheck<StreamRecord<T>>[] outputWithRecordsCountChecks;
     protected final Output<StreamRecord<T>>[] outputs;
     private final Random random = new XORShiftRandom();
     private final WatermarkGauge watermarkGauge = new WatermarkGauge();
+    protected final Counter numRecordsOutForTask;
 
-    public BroadcastingOutputCollector(Output<StreamRecord<T>>[] outputs) {
+    public BroadcastingOutputCollector(
+            OutputWithRecordsCountCheck<StreamRecord<T>>[] outputWithRecordsCountChecks,
+            Output<StreamRecord<T>>[] outputs,
+            Counter numRecordsOutForTask) {
         this.outputs = outputs;
+        this.outputWithRecordsCountChecks = outputWithRecordsCountChecks;
```

Comment:
The reason why `output` is not removed is that some methods are still on the `Output` interface, such as `emitLatencyMarker`.